loadout items now spawn in lockers in dorms + persistent GM medals !!!NEED TESTING!!!

AndroBetel opened the pull request:
i'm not smart enough to set up a database for medals to spawn but i hope it works out, still needs testing

FslashN reviewed (changes requested):
Very soulful. Small issue:
Squad sergeants don't seem to spawn with a personal locker from what we were able to discover.

Also consider combining the cryo lockers with the personal lockers, so marines get all of their gear from their personal locker instead of two different lockers.

AndroBetel commented:
sgt and pltsgt lockers moved to dorms
removed cryo lockers, added cryo gear to personal lockers
"persistent" medals tested and they should work, might need a bit of tweaking, but for now they're fine

morrowwolf commented on code/modules/gear_presets/_select_equipment.dm:
```diff
@@ -153,7 +153,10 @@
 	load_id(new_human, mob_client)
 	load_status(new_human, mob_client)
-	load_vanity(new_human, mob_client)
+	if(is_ground_level(new_human.z))
```
Probably want is_ship_level() (if we have that, otherwise you'll need to make it)

AndroBetel replied:
the problem is that sometimes marines spawn in cryosleep pods and that fucks up this check, and props never spawn inside other atoms, so thats why i check for ground level

morrowwolf replied:
This is easily checked via a quick get_turf() before checking the level

morrowwolf commented on code/modules/gear_presets/_select_equipment.dm:
```diff
@@ -246,6 +249,96 @@
+
+/datum/equipment_preset/proc/spawn_vanity_in_personal_lockers(mob/living/carbon/human/new_human, client/mob_client)
+	var/obj/structure/closet/secure_closet/marine_personal/closet_to_spawn_in
+	if(!new_human.client || !new_human.client.prefs || !new_human.client.prefs.gear)
```
This can be turned into just !new_human.client?.prefs?.gear iirc

morrowwolf commented on code/modules/gear_presets/_select_equipment.dm:
```diff
@@ -246,6 +249,96 @@
+	if(new_human.disabilities & NEARSIGHTED)
+		new /obj/item/clothing/glasses/regular(closet_to_spawn_in)
+
+	for(var/datum/view_record/medal_view/medal as anything in DB_VIEW(/datum/view_record/medal_view, DB_COMP("player_id", DB_EQUALS, new_human.client.player_data.id)))
```
This whole thing is awesome, big ups

private-tristan commented:
Vendors (specifically the medic one) will spawn uniforms with payment pins.
Medic Vendor distributing clothing (other than helmet/sterile gloves) should probably be removed so that you use the personal locker.
medic's personal locker will contain a normal uniform, and not the corpsman variant.
